src/app/file-panel.js

```diff
@@ -108,14 +112,16 @@ function filepanel (appAPI, files) {
 
   var events = new EventManager()
   var element = template()
-  // TODO please do not add custom javascript objects, which have no
-  // relation to the DOM to DOM nodes
-  element.events = events
+
   fileExplorer.events.register('focus', function (path) {
     appAPI.switchToFile(path)
   })
 
-  return element
+  var api = { events: events }
+
+  container.appendChild(element)
+
+  return api
```

Comment:
The `new` keyword doesn't care whether there is an explicit return or not actually. That's part of the spec. I personally prefer explicit over implicit.

I will change it to `this.events = events` but can personally not recommend it.
In JavaScript I consider classes as a legacy that is less flexible than more modern approaches, but if this style is preferred i will stick to it.

Here some links:
* https://twitter.com/peitor/status/845178317233807360
* https://medium.com/javascript-scene/common-misconceptions-about-inheritance-in-javascript-d5d9bab29b0a
* https://medium.com/javascript-scene/javascript-factory-functions-vs-constructor-functions-vs-classes-2f22ceddf33e
* http://wiki.c2.com/?ArgumentsAgainstOop